.travis.yml

```diff
@@ -37,16 +35,15 @@ deploy:
     on:
       tags: true
       condition: $TRAVIS_TAG =~ $PROD_TAG_REGEXP
```

Comment:
Hmm... we might have a problem here.

This deployment stuff deals with uploading the pattern library to output to AWS for production releases. The idea is that whenever we publish a new NPM package version, we _also_ update what you see at [`http://style.buildit.digital/`](http://style.buildit.digital/) so that both are always in sync.

Currently, the condition that triggers this deployment is a git tag with that matches the SemVer `vX.Y.Z` format.

This worked so far, because we would manually tag the commit we wanted to release (typically using `npm version ..` so that `package.json` also got updated at the same time). When we pushed that commit to master, Travis would run, see the new tag and know to publish both the pattern library site and the NPM package.

Now, we're removing that step, since `semantic-release` will deal with making a changelog, bumping the version and publishing the NPM package. And it will be Travis executing that - not a developer.

*If* is also commits a tag (which I suspect it will because according to the docs it uses `npm version ...` under the hood), the I suppose our pattern library publish should also work as long as we ensure that it runs _after_ `semantic-release`. I'm not 100% sure about that though - because I'm not sure at which point Travis checks the `on:` conditions. If it only does so at the start of the job, then we'll need to find some other way.

Either way, this is something we need to address. Perhaps something to try out in your fork?

Reply:
Relatedly, if `semantic-release` does indeed add a git tag - does it then also push it back to our Github repo? Or does it only exist in Travis' local clone of our repo and then vanish when the Travis job completes?

If possible, I'd quite like it to do so. I think it's helpful to have whichever commits were released tagged. That way we can easily check out the code as it was for a particular version. I can see that being helpful if we ever need to reproduce a bug on an older version. Or are trying to figure in which version a specific change was introduced.

Reply:
Semantic Release auto tags the code and runs a separate branch push to update the git tags. So I think we're safe here. If I'm not let me know (screenshots of example build below).

<img width="1680" alt="screen shot 2019-02-27 at 8 22 55 am" src="https://user-images.githubusercontent.com/643307/53501146-eedd4880-3a68-11e9-993e-25d2b9249697.png">

Reply:
Sounds good.

Ok, let's proceed with this and try it out. Not really sure how we can test whether the pattern library deployment works when we want it to, without doing this for real. :-)

Worst case, if there is some issue, we'll resolve it then. It's not the end of the world if your published pattern library is out of sync for a release or two.

Reply:
I was thinking the same, should be a 90% chance it works.